From b6f743d1601a88a1b53f8d8a3b8bfd368883917a Mon Sep 17 00:00:00 2001
From: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
Date: Fri, 8 Dec 2023 17:05:58 +0530
Subject: [PATCH] Fix crash when bt headset connects on aux BT

Fix null pointer access in handle_rc_ctrl_features

There is a chance the callback gets invoked right after a AVRC
disconnection and bt_rc_ctrl_callbacks has been cleared.

Cherry pick from AOSP:
commit id: 722854df05ddad0567f5c30db2491afc90d15228
https://android-review.googlesource.com/c/platform/packages/modules/Bluetooth/+/2499218

Tracked-On: OAM-114416
Signed-off-by: Ugo Yu <ugoyu@google.com>
Signed-off-by: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
---
 btif/src/btif_rc.cc | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/btif/src/btif_rc.cc b/btif/src/btif_rc.cc
index 31a0cb98e..2d63a9195 100644
--- a/btif/src/btif_rc.cc
+++ b/btif/src/btif_rc.cc
@@ -730,12 +730,13 @@ void handle_rc_connect(tBTA_AV_RC_OPEN* p_rc_open) {
     do_in_jni_thread(FROM_HERE,
                      base::Bind(bt_rc_ctrl_callbacks->connection_state_cb, true,
                                 false, p_dev->rc_addr));
-  }
-  /* report connection state if remote device is AVRCP target */
-  handle_rc_ctrl_features(p_dev);
 
-  /* report psm if remote device is AVRCP target */
-  handle_rc_ctrl_psm(p_dev);
+    /* report connection state if remote device is AVRCP target */
+    handle_rc_ctrl_features(p_dev);
+
+    /* report psm if remote device is AVRCP target */
+    handle_rc_ctrl_psm(p_dev);
+  }
 }
 
 /***************************************************************************
-- 
2.17.1

